lib/pure/json.nim

```diff
@@ -366,6 +367,12 @@ proc `%`*(o: enum): JsonNode =
   ## string. Creates a new ``JString JsonNode``.
   result = %($o)
 
+template identOrValue*(x: untyped): untyped =
+  when not declaredInScope(x):
```

Comment:
I don't understand this logic.

Reply:
The idea is a template that's injected into the calling scope of the macro so that we can check whether some symbol is already declared as a variable or not.

Reply:
Ok, that's a very good idea. But it is too clever and will cause problems.